src/aw-client.ts

```diff
@@ -138,23 +138,16 @@ export class AWClient {
         return this.req.get("/0/buckets/" + bucketId + "/events/count", { params });
     }
 
-    public async insertEvent(bucketId: string, event: IEvent): Promise<IEvent> {
-        return this.insertEvents(bucketId, [event]).then(events => events[0]);
+    public async insertEvent(bucketId: string, event: IEvent): Promise<void> {
+        return this.insertEvents(bucketId, [event]);
```

Comment:
I'm fine with this. But isn't there any way to get at least the ID of the inserted events back without a performance hit?

Reply:
The ID is decided when it is commited to the database, which is not until after the transaction is commited.

Or so I thought, I googled it and found this

https://stackoverflow.com/a/620831/2127148

So apparently the auto generated primary key is decided when you flush to disk, not when the transaction is commited.

So I guess that it would worsen I/O performance anyway...

Reply:
Ok, now I'm confused as I realized that SQLIte apparently supports the lastrowid call even when using transactions...
So it should be possible to return the event when we only insert one event.